reana_client/cli/common.py

```diff
@@ -0,0 +1,125 @@
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA client cli functionality shared by multiple cli commands."""
+
+import click
+import logging
+import os
+import traceback
+
+
+def get_current_click_command(ctx):
+    """Return command (and subcommands) that were used to create given context.
+
+    For example, a context that was created by a command
+    '<executable name> code upload', would return 'code.upload'.
+
+    """
+    command = []
+    while ctx.parent:
+        command.append(ctx.info_name)
+        ctx = ctx.parent
+    return '.'.join(str(x) for x in reversed(command))
+
+
+def upload_files_to_server(user, organization, workflow, filenames, client):
+    """Upload files to REANA-Server.
+
+    Shared e.g. by `code upload` and `inputs upload`.
+
+    :param user: User ID
+    :param organization: Organization ID
+    :param workflow: ID of that Workflow whose workspace should be
+        used to store the files.
+    :param filenames: Absolute filepath(s) of files to be uploaded.
+    :param client: Instance of Bravado Client.
+    """
+    if workflow:
+        logging.info('Workflow "{}" selected'.format(workflow))
+        if filenames:
+            for filename in filenames:
+                try:
+                    # Check if input is a folder and upload everything
+                    # including subdirectories.
+                    if os.path.isdir(filename):
+                        # TODO: Implement actual folder upload (with globbing)
+                        logging.debug('{} is a directory.'.format(filename))
+                        logging.debug('Directory upload not supported yet.')
+                        click.echo(
+                            click.style(
+                                'Skipping upload of folder \'{}\' .'
+                                'Directory upload not supported yet.'.
+                                format(filename),
+                                fg='red'))
+
+                    # Check if input is an absolute path and the upload file.
+                    # Shell and Click should resolve the input to an absolute
+                    # filepath.
+                    elif os.path.isabs(filename):
+                        with click.open_file(filename) as f:
+                            name = os.path.basename(f.name)
+                            logging.debug('{} is an absolute filepath.'
+                                          .format(os.path.basename(name)))
+                            click.echo('Uploading \'{}\' ...'.format(name))
+
+                            ctx = click.get_current_context()
+                            if 'code upload' in ctx.command_path:
```

Comment:
I find this a bit fragile, if `command_path` changes the format at come point this would break, could we maybe have a new parameter which would represent the type instead?